### PR TITLE
add getting contact by field

### PR DESCRIPTION
**Pull request by rudigiesler**
The functionality for getting a contact by a field (eg. MSISDN) is now implemented in the contacts API, we should now implement this functionality in the client.

### CODE REVIEW

**Comment by rudigiesler** [comment]
Tests failing because pep8 is failing on things that this PR didn't touch. Looking into it...

**Comment by rudigiesler** [comment]
Seems a newer version of pep8 that checks for more things. Updated locally and it showed the same pep8 errors. pep8 errors are now fixed.

**Comment by rudigiesler** [comment]
Ready for a first review.

File: go_http/contacts.py [hunk 1]
```diff
@@ -95,6 +97,19 @@ def get_contact(self, contact_key):
         """
         return self._api_request("GET", "contacts", contact_key)
 
+    def get_contact_from_field(self, field, value):
```
Comment (hodgestar) [inline go_http/contacts.py]: I wondering whether we shouldn't overload ``.get_contact()`` and do something like ``.get_contact(*args, **kw)`` and then:
```
if not kw and len(args) == 1:
    return self._contact_by_key(args[0])
elif len(kw) == 1 and not args:
    field, value = kw.items()[0]
    return self._contact_by_field(field, value)
raise ValueError(".contact() may either be called as .get_contact(contact_key) or .get_contact(field=value)")
```
Thoughts?

File: go_http/contacts.py [hunk 2]
```diff
@@ -95,6 +97,19 @@ def get_contact(self, contact_key):
...
+        """
+        Get a contact given a field and a value for that field.
+
+        :param str field:
+            Field that is searched on (eg. ``MSISDN``)
```
Comment (hodgestar) [inline go_http/contacts.py]: We should probably list more of the fields here (perhaps just the most common ones -- `msisdn`, `twitter_handle`).

File: go_http/contacts.py [hunk 3]
```diff
@@ -95,6 +97,19 @@ def get_contact(self, contact_key):
...
+        :param str value:
+            Value that the field must match (eg. ``+12345``)
```
Comment (hodgestar) [inline go_http/contacts.py]: Missing `.` in `eg.`.

**Comment by hodgestar** [comment]
Left one suggestion and some minor comments. Looks good though.

File: go_http/contacts.py (hunk 1 above)
Comment (rudigiesler) [inline go_http/contacts.py]: That seems good, I'll implement that.

**Comment by rudigiesler** [comment]
@hodgestar Changes implemented. Ready for another review.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**